src/message.c

```diff
@@ -1225,7 +1241,7 @@ int qd_message_get_phase_annotation(const qd_message_t *in_msg)
 void qd_message_set_stream_annotation(qd_message_t *in_msg, bool stream)
 {
     qd_message_pvt_t *msg = (qd_message_pvt_t*) in_msg;
-    msg->content->ma_stream = stream;
+    sys_atomic_set(&msg->content->ma_stream, (stream ? 1 : 0));
```

Comment:
just a thought. Instead of calling sys_atomic_set can we do the following - 
if (stream) {
SET_ATOMIC_FLAG(&msg->content->ma_stream)
}
else {
CLEAR_ATOMIC_FLAG(&msg->content->ma_stream)
}